### ISSUE TITLE
Nested fat arrow closures cause false positive for UndefinedVariable

### ISSUE THREAD

**Issue by stefanfisk** [sirbrillig/phpcs-variable-analysis#329]
Here's a simple example that generates `UndefinedVariable` for the second `$item` in the array.

```php
$fn = fn($item) => [
    $item,
    fn($arg) => $arg,
    $item,
];
```

This works fine.

```php
$fn = function($item) {
    return [
        $item,
        fn($arg) => $arg,
        $item,
    ];
}
```

I am using v2.11.19 but the issue has been around for years, I've just never gotten around to opening an issue.

**Comment by sirbrillig** [comment]
Thank you for the report! I don't think I ever did enough testing with nested fat arrow functions.

**Comment by sirbrillig** [comment]
Should be fixed by https://github.com/sirbrillig/phpcs-variable-analysis/pull/334
